File: tests/python/mkdata.py
#!/usr/bin/python2.5

# Load the python module.
import sys
import os
sys.path[0] = ''
os.chdir ("../../lib")
import cdbpy
os.chdir ("../tests/python")

numItems = 0
if len (sys.argv) == 2:
  numItems = int(sys.argv[1])
else:
  numItems = 5000

fp = open ("testAdd.txt", "w", 0644)

i = 0
while (i < numItems):
  LName = cdbpy.dbtestupperstring (0, 64)
  file.write (fp, LName)
  file.write (fp, ":")
  SSNumber = cdbpy.dbtestnumber (9, 0)
  file.write (fp, SSNumber)
  file.write (fp, ":")
  FName = cdbpy.dbtestupperstring (0, 34)
  file.write (fp, FName)
  file.write (fp, ":")
  StreetAddress = cdbpy.dbteststring (0, 34)
  file.write (fp, StreetAddress)
  file.write (fp, ":")
  City = cdbpy.dbteststring (45, 0)
  file.write (fp, City)
  file.write (fp, ":")
  State = "NM"
  file.write (fp, State)
  file.write (fp, ":")
  Zip = cdbpy.dbtestnumber (10, 0)
  file.write (fp, Zip)
  file.write (fp, ":")
  AnnualSalary = cdbpy.dbtestnumber (7, 0)
  file.write (fp, AnnualSalary)
  file.write (fp, ".00")
  file.write (fp, ":")
  HomeOwner = "Y"
  file.write (fp, HomeOwner)
  file.write (fp, ":")
  DateApplied = cdbpy.dbtestnumber (8, 0)
  file.write (fp, DateApplied)
  file.write (fp, ":")
  LastUpdated = cdbpy.dbteststring (26, 0)
  file.write (fp, LastUpdated)
  file.write (fp, "\n")
  i += 1

file.close (fp)
sys.exit (0)

File: tests/c/mkdata.c
/* Source File: mkdata.c */

#ifdef __CINT__
#include "cint_types.h"
#endif

#ifndef __CINT__
#include <stdio.h>
#include <interface.h>
#define MAXRECORDS 1000000
#endif

#define DEFAULT_NUMBER 10000

int main (int argc, char *argv[])
{
  char *str;
  FILE *fp;
  int counter=0;
  int count = DEFAULT_NUMBER;
  int smalldate;

  if (argc > 1) {
    count = atoi (argv[1]);
    if (count < 0 || count > MAXRECORDS) {
      count = 10000;
    }
  }

  fp = fopen ("testAdd.txt", "w+b");
  while (counter++ < count) {
    /* LName */
    str = dbtestupperstring (0, 64);
    if (str == 0) {
      printf ("\n\n***Error: LName: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s:", str);
    /* SSNumber */
    str = dbtestnumber (9, 0);
    if (str == 0) {
      printf ("\n\n***Error: SSNumber: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s:", str);
    /* FName */
    str = dbtestupperstring (0, 34);
    if (str == 0) {
      printf ("\n\n***Error: FName: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s:", str);
    /* StreetAddress */
    str = dbteststring (75, 0);
    if (str == 0) {
      printf ("\n\n***Error: StreetAddress: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s:", str);
    /* City */
    str = dbteststring (45, 0);
    if (str == 0) {
      printf ("\n\n***Error: City: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s:", str);
    /* State */
    fprintf(fp, "%s:", "NM");
    /* Zip */
    str = dbtestnumber (10, 0);
    if (str == 0) {
      printf ("\n\n***Error: Zip: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s:", str);
    /* AnnualSalary */
    str = dbtestnumber (5, 0);
    if (str == 0) {
      printf ("\n\n***Error: FName: %s\n", cdberror);
      return _ERROR_;
    }
    fprintf(fp, "%s.00:", str);
    /* HomeOwner */
    fprintf (fp, "Y:");
    /* DateApplied */
    str = dbtestnumber (6, 0);
    if (str == 0) {
      printf ("\n\n***Error: FName: %s\n", cdberror);
      return _ERROR_;
    }
    smalldate = atoi (str);
    if (smalldate < 0 || smalldate > 991231)
      smalldate = 651010; /* arbitrary birthdate */
    smalldate += 19000000;
    fprintf(fp, "%d:", smalldate);
    /* LastUpdated */
    str = dbtestnumber (4, 0);
    if (str == 0) {
      printf ("\n\n***Error: FName: %s\n", cdberror);
      return _ERROR_;
    }
    smalldate = atoi (str);
    if (smalldate < 0 || smalldate > 1231)
      smalldate = 1010;
    smalldate += 20020000;
    fprintf(fp, "%d\n", smalldate);
  }
  fclose (fp);
  return 0;
}
